src/main/java/shadows/apotheosis/advancements/ExtendedInvTrigger.java

```diff
@@ -58,7 +58,7 @@ ItemPredicate[] deserializeApoth(JsonObject json) {
         }
         if ("rarity".equals(type)) {
             LootRarity rarity = LootRarity.byId(json.get("rarity").getAsString().toLowerCase(Locale.ROOT));
-            return new ItemPredicate[] { new TrueItemPredicate(s -> AffixHelper.getRarity(s) == rarity) };
+            return new ItemPredicate[] { new TrueItemPredicate(s -> !AffixHelper.getAffixes(s).isEmpty() && AffixHelper.getRarity(s) == rarity) };
```

Comment:
This change is unnecessary since affixes and gems use different nbt formats in 1.19